File: src/main/java/rustleund/fightingfantasy/framework/closures/impl/TestEnemyStatPredicate.java
package rustleund.fightingfantasy.framework.closures.impl;

import org.w3c.dom.Element;

import rustleund.fightingfantasy.framework.base.AbstractEntityState;
import rustleund.fightingfantasy.framework.base.GameState;

public class TestEnemyStatPredicate extends TestStatPredicate {

	private int enemyId;

	public TestEnemyStatPredicate(Element element) {
		super(element);
		this.enemyId = Integer.valueOf(element.getAttribute("enemyId"));
	}

	@Override
	protected AbstractEntityState getEntityStateToTest(GameState gameState) {
		return gameState.getBattleState().getEnemies().getEnemies().get(enemyId);
	}
}

File: src/main/java/rustleund/fightingfantasy/framework/closures/impl/TestEnemyStatPredicate.kt
package rustleund.fightingfantasy.framework.closures.impl

import org.w3c.dom.Element
import rustleund.fightingfantasy.framework.base.GameState
import java.util.function.Predicate

private fun Element.enemyIdx() = this.getAttribute("enemyIdx").toInt()

class TestEnemyStatPredicate(element: Element) : Predicate<GameState> by
TestStatPredicate(
    element,
    { it.battleState?.enemies?.enemies?.get(element.enemyIdx()) ?: throw IllegalStateException("No battle in progress") },
    { it.battleState?.currentAttackStrengths?.getEnemyAttackStrength(element.enemyIdx()) }
)
